### PR TITLE
Use a WeakMap and fix memory leak

### PR DESCRIPTION
**Pull request by mjc-gh**
Pretty sure the current code has a memory leak in it since nothing is removing DOM elements from the `elements` array. Using a `WeakMap` instead of an array should properly deal with this leak.

Thanks!

### CODE REVIEW

**Comment by zzarcon** [comment]
Hi @mikeycgto ! Changes look awesome, I think its a good idea to use WeakMap.

Do you what happened in the **dist** file? it shows -1,157 changes. Happy to get it merged after we solve that :)

**Comment by mjc-gh** [comment]
Sorry about that! My `npm run build` initially failed because I was missing the bower package, hence bad diff.

This should be good now. Thanks for the quick feedback!

File: react-keypress.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 var Keypress = require("./bower_components/Keypress/keypress.js");
-var elements = [];
+var elements = new WeakMap;
```
Comment (zzarcon) [inline react-keypress.js]: I believe you need to do ```new WeakMap()```. Is there any reason to not use ```()```?

**Review by zzarcon — CHANGES_REQUESTED** [review]
Just added a minor comment, have a look and lets get it merged! :)

**Comment by zzarcon** [comment]
@mikeycgto will you be interested of being part of the repository? I can add you as contributor and give you publish rights on npm as well.

I believe there are some things we can improve if you are motivated:

* Remove **/dist** from Git.
* Remove Bower
* Introduce Webpack
* Use new Javascript features

What do you think? :)

**Comment by mjc-gh** [comment]
Yeah that would be great! We can introduce babel to and use es6 syntax. I'd also like to write some simple tests as well.

File: react-keypress.js (hunk 1 above)
Comment (mjc-gh) [reply react-keypress.js]: Just a small style thing I do with initializing new objects without any args. Amended the commit to include them :+1:

**Comment by zzarcon** [comment]
Awesome! You are collaborator here and on npm now 🍾 .

I will create some issues now to track the stuff we want to do :)

**Pull request merged**